tests/TestCase/Utility/TextTest.php

```diff
@@ -1879,6 +1879,22 @@ public function slugInputProvider()
                 'cl#e|an(me.jpg', ['preserve' => '.'],
                 'cl-e-an-me.jpg',
             ],
+            [
+                'Foo Bar: Not just for breakfast any-more', ['preserve' => ' '],
+                'Foo Bar- Not just for breakfast any-more',
+            ],
+            [
+                'Foo Bar: Not just for (breakfast) any-more', ['preserve' => ' ()'],
+                'Foo Bar- Not just for (breakfast) any-more',
+            ],
+            [
+                'Foo Bar: Not just for breakfast any-more', ['replacement' => null],
+                'FooBarNotjustforbreakfastanymore',
+            ],
+            [
+                'Foo Bar: Not just for breakfast any-more', ['replacement' => false],
```

Comment:
Since you are using `strlen()` setting `replacement` to `false` will generate type error when this is merged into 4.x.

Reply:
Nice catch. I changed the check once more to `is_string` instead of `isset`.